runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZonePlugin.java

```diff
@@ -0,0 +1,186 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.nightmarezone;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import net.runelite.api.*;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.Notifier;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.ClientUI;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.inject.Inject;
+
+@PluginDescriptor(
+		name = "Nightmare Zone Plugin"
+)
+public class NightmareZonePlugin extends Plugin
+{
+	private static final Point NMZ_EXIT = new Point(2608, 3115);
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private ClientUI gui;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private NightmareZoneConfig config;
+
+	@Inject
+	private NightmareZoneOverlay overlay;
+
+	@Override
+	public void configure(Binder binder)
+	{
+		binder.bind(NightmareZoneOverlay.class);
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	// This starts as true since you need to get
+	// above the threshold before sending notifications
+	private boolean absorptionNotificationSend = true;
+
+	private boolean inNightmareZone = false;
+
+	@Provides
+	NightmareZoneConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(NightmareZoneConfig.class);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.enabled())
+		{
+			return;
+		}
+		if (config.absorptionNotification())
+		{
+			checkAbsorption();
+		}
+		if (inNightmareZone)
+		{
+			Player player = client.getLocalPlayer();
+			Point playerLocation = player.getWorldLocation();
+			// It's not possible to teleport out of nmz, so when you exit/die
+			// you will always spawn right by Dominic Onion
+			if (playerLocation.equals(NMZ_EXIT))
+			{
+				inNightmareZone = false;
```

Comment:
This is probably not the most robust way to do this. People can hop accounts, or even game types, and it would lose track. There was some discussion earlier on using client.getMapRegions() to detect which region you are in, for when you are in instances.